### ISSUE TITLE
Add an option to exclude csproj files from showing under "Open a recent item"

### ISSUE THREAD

**Issue by MagicAndre1981** [luminous-software/start-page-plus#46]
### Installed product versions
- Visual Studio: 2019 16.8.2
- This extension: 1.0.1

### Description

add an option to exclude csproj files from showing under recent items

### Steps to recreate / Current behavior
1. open sln
2. add an existing csproj to the sln 
3. close sln
4. the csproj now show up on recent

### Expected behavior

recent items should not show csproj because I always open the sln for the project. For people who like the old way an option should be added to control which files get added to `recent items`.

**Comment by yannduran** [comment]
Hmm, I haven't actually noticed that happening. It's certainly not behavior that I've introduced, and I certainly wouldn't do that without a setting to revert to the previous behavior. I **hate** when Microsoft does that, so I wouldn't do that to my users!

In _v1.0_, I introduced a setting called _Show Extensions_ to show or hide project/solution extensions, but I haven't even released any changes since then. The default setting is `false` (which hides the extensions).

A couple of questions:

1. Is the icon a green _C# icon_ or a _solution icon_ (rectangle with a purple VS icon in the bottom left corner)?

2. If it's not a _C# icon_, is it at all possible that you've opened a project file from outside of _Start Page+_? Maybe using Microsoft's _Start Window_, or by clicking on a project file directly?

3. Do all solutions show up as projects? You can toggle the _Show Extensions_ setting to see if that makes it any easier to tell.

4. What do you see if you open the _Start Window_? I'm wondering if Microsoft may have made an internal change in a recent update that is affecting the list.

If none of that helps, let me know and I'll have a closer look at what's going on.

**Comment by MagicAndre1981** [comment]
I see the green csproj icons and I only want to have sln shown. I noticed that Start Window also shows csprojs, but this only happens if I add csprojs to a sln.

I think this is the default behavior of Visual Studio, it adds added csproj to the recent items list (check it with oldest VS I have (VS2012)).

**Comment by yannduran** [comment]
>"_I think this is the default behavior of Visual Studio, it adds added csproj to the recent items list_"

Yes, I agree. It's not something that _Start Page+_ controls, it just uses Visual Studio's recent items list.

My advice would be to just delete the csproj entries that you don't want to see. If you see them pop up again after deleting them, it'd be interesting to know what **exact** actions makes them appear again. 

But I doubt that I'll be able to do much about them reappearing, unless I added a filter to **only** show solutions (feel free to add that as a feature request, but no promises OK?). But clicking on a project opens the solution anyway, so is it really a huge problem?

If you're happy to do so, I'll leave it to close this issue. Any further discussion should be in the feature request (if you decide to open one).

**Comment by MagicAndre1981** [comment]
> But I doubt that I'll be able to do much about them reappearing, unless I added a filter to **only** show solutions (feel free to add that as a feature request, but no promises OK?). 

this is my feature request, try to filter them and only show sln files, no csproj.

**Comment by zippy72** [comment]
I'd love this, personally. I've actually posted a similar issue before on the Visual Studio issue tracker to ask for the recent items menu to allow this.
